src/consensus/validation.h

```diff
@@ -88,4 +90,151 @@ class CValidationState
     void SetDebugMessage(const std::string &s) { strDebugMessage = s; }
 };
 
+class CInputData
```

Comment:
please doxygen comment all classes and APIs

Reply:
will do